python/src/scipp/compat/mantid.py

```diff
@@ -261,6 +288,30 @@ def convertEventWorkspace_to_dataarray(ws, load_pulse_times):
     return sc.DataArray(**coords_labs_data)
 
 
+def convertMDHistoWorkspace_to_dataset(md_histo):
+    ndims = md_histo.getNumDims()
+    if ndims > 3:
+        raise RuntimeError("Converter cannot process md histo workspace input "
+                           "with > 3 dimensions. Input has {}.".format(ndims))
+    coords = dict()
+    dims_used = []
+    for i in range(ndims):
+        dim = md_histo.getDimension(i)
+        frame = dim.getMDFrame()
+        sc_dim = md_dimension(dim, i)
+        coords[sc_dim] = sc.Variable(dims=[sc_dim],
+                                     values=np.linspace(
+                                         dim.getMinimum(), dim.getMaximum(),
+                                         dim.getNBins()),
+                                     unit=md_unit(frame))
+        dims_used.append(sc_dim)
```

Comment:
From what I can tell, in `MDHistoWorkspace` the first index is the fastest (inner) index in the array. Scipp uses numpy convention, so I think you need to reverse the order of dimension labels? You should be able to see this in your test if you use a rectangular volume?

Reply:
I extract the data `getSignalArray()` as a numpy array (should be numpy convention), so I think this should be OK as is. I've extended the tests a little they already dealt with different bin numbers for each dimension.

Reply:
Well, I am not questioning that the array is correct (we do *not* want to transpose it, this would be very costly)). My point is that the dimension labels may be reversed.